src/passes/OptimizeInstructions.cpp

```diff
@@ -820,6 +736,73 @@ struct OptimizeInstructions
           }
         }
       }
+      if (auto* condition = select->condition->dynCast<Unary>()) {
+        if (condition->op == EqZInt32) {
+          // flip select to remove eqz, if we can reorder
+          EffectAnalyzer ifTrue(getPassOptions(), features, select->ifTrue);
+          EffectAnalyzer ifFalse(getPassOptions(), features, select->ifFalse);
+          if (!ifTrue.invalidates(ifFalse)) {
+            select->condition = condition->value;
+            std::swap(select->ifTrue, select->ifFalse);
+          }
+        }
+      }
+      if (auto* constTrue = select->ifTrue->dynCast<Const>()) {
+        if (auto* constFalse = select->ifFalse->dynCast<Const>()) {
+          Builder builder(*getModule());
+          bool isPartially64bits =
```

Comment:
what does "partially" mean here? It's not clear enough to me so many it should be renamed.

also a select condition is always an i32, so there is no need to check it (unless you are checking it's not unreachable?)

Reply:
Yeah, after recent changes it's not need any more. Fixed